### PR TITLE
bpo-37840: Fix handling of negative indices in bytearray_getitem()

### PR DESCRIPTION
**Pull request by sir-sigurd**
https://bugs.python.org/issue37840

### CODE REVIEW

File: Lib/test/test_bytes.py [hunk 1]
```diff
@@ -962,6 +962,10 @@ def test_translate(self):
         c = b.translate(None, delete=b'e')
         self.assertEqual(c, b'hllo')
 
+    def test_sq_item(self):
+        _testcapi = test.support.import_module('_testcapi')
+        self.assertRaises(IndexError, _testcapi.sequence_getitem, self.type2test(1), -2)
```
Comment (ZackerySpytz) [inline Lib/test/test_bytes.py]: Please wrap at 79 chars.

File: Misc/NEWS.d/next/Core and Builtins/2019-08-13-18-05-20.bpo-37840.elLCci.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Fix handling of negative indices in :c:member:`~PySequenceMethods.sq_item`
+of `bytearray`. Patch by Sergey Fedoseev.
```
Comment (ZackerySpytz) [inline Misc/NEWS.d/next/Core and Builtins/2019-08-13-18-05-20.bpo-37840.elLCci.rst]:
```suggestion
of :class:`bytearray`. Patch by Sergey Fedoseev.
```

File: Modules/_testcapimodule.c [hunk 3]
```diff
@@ -5095,6 +5095,18 @@ test_write_unraisable_exc(PyObject *self, PyObject *args)
 }
 
 
+static PyObject*
```
Comment (ZackerySpytz) [inline Modules/_testcapimodule.c]:
```suggestion
static PyObject *
```

File: Misc/NEWS.d/next/Core and Builtins/2019-08-13-18-05-20.bpo-37840.elLCci.rst [hunk 4]
```diff
@@ -0,0 +1,2 @@
+Fix handling of negative indices in :c:member:`~PySequenceMethods.sq_item`
```
Comment (epicfaace) [inline Misc/NEWS.d/next/Core and Builtins/2019-08-13-18-05-20.bpo-37840.elLCci.rst:1]: Should this be PySequenceMethods.sq_item or PySequence_GetItem?

File: Lib/test/test_bytes.py [hunk 5]
```diff
@@ -962,6 +962,11 @@ def test_translate(self):
         c = b.translate(None, delete=b'e')
         self.assertEqual(c, b'hllo')
 
+    def test_sq_item(self):
```
Comment (epicfaace) [inline Lib/test/test_bytes.py:965]: Can you add some tests that call sequence_getitem with a non-negative index as well?

File: Misc/NEWS.d/next/Core and Builtins/2019-08-13-18-05-20.bpo-37840.elLCci.rst (hunk 4 above)
Comment (sir-sigurd) [reply Misc/NEWS.d/next/Core and Builtins/2019-08-13-18-05-20.bpo-37840.elLCci.rst:1]: PySequenceMethods.sq_item

File: Lib/test/test_bytes.py (hunk 5 above)
Comment (epicfaace) [reply Lib/test/test_bytes.py]: I meant, could you add a test that calls sequence_getitem successfully? (to make sure the successful behavior is not impacted)

Comment (epicfaace) [reply Lib/test/test_bytes.py:965]: I meant, could you add a test that calls sequence_getitem successfully? (to make sure the successful behavior is not impacted)

**Review by epicfaace — CHANGES_REQUESTED** [review]

File: Lib/test/test_bytes.py [hunk 6]
```diff
@@ -962,6 +962,15 @@ def test_translate(self):
...
+        _testcapi = test.support.import_module('_testcapi')
+        obj = self.type2test(1)
+        with self.assertRaises(IndexError):
+            _testcapi.sequence_getitem(obj, -2)
+        with self.assertRaises(IndexError):
+            _testcapi.sequence_getitem(obj, 1)
+        _testcapi.sequence_getitem(obj, 0)
```
Comment (epicfaace) [inline Lib/test/test_bytes.py]: Can you make sure this returns the right thing (should return `1` I think?)

**Review by epicfaace — APPROVED** [review]

**Comment by Yhg1s** [comment]
(closing and reopening to rerun the presubmits...)

**Comment by sir-sigurd** [comment]
@Yhg1s shouldn't this be backported?

**Pull request merged**